implement crayon rotation

neomoth opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Implemented the ability to rotate decals placed by crayons.
Partial credit to [themias](https://github.com/themias), turns out this was a duplicate of their PR!! Used their code as reference for adding my own implementation of the decal preview ghost. Didn't know you could do coauthor stuff with git before my initial commit, and I have no idea if it's possible to add that after the fact so I just edited this PR to give attribution here. Also added them as an author on the changelog.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Under normal circumstances, attempting to place down a decal via crayon will always result in the orientation of the decal matching whatever the "default" rotation is for the camera for whatever it is that you're standing on. (e.g. take crayon, step onto a surface, press Numpad8 to orient camera to default rotation, whatever you draw will be upright from this view)
This would make it rather difficult to write messages in certain spaces unless you want to do it manually with the brushes. To fix this, a "Glyph rotation" box has been added.

## Technical details
<!-- Summary of code changes for easier review. -->
Sort of just copied what the crayon does for setting the color of the crayon (RGB sliders) and copied the implementation of the rotation box from another UI. This involved adding a `CrayonRotationMessage` to `SharedCrayonComponent.cs` in order to relay the rotation info from client to server. Of course, changes to `CrayonWindow.xaml` and `CrayonWindow.xaml.cs` were also made, and other small additions were made in other various crayon related files in order to ensure rotation is saved properly on the server and relayed back to the client when it next opens the crayon UI.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
<img width="464" height="510" alt="image" src="https://github.com/user-attachments/assets/6976bb2e-69cd-49cf-9d1b-4a254b4be6fc" />
<img width="913" height="505" alt="image" src="https://github.com/user-attachments/assets/ee540369-35c9-4be1-a5c5-426610db4099" />
<img width="418" height="335" alt="image" src="https://github.com/user-attachments/assets/edb53a78-841a-4ebe-aa13-5bc4a2b94c71" />
<img width="115" height="125" alt="image" src="https://github.com/user-attachments/assets/651c0de3-cbcd-4fe7-8d77-da537b8203db" />

https://github.com/user-attachments/assets/b0a1754c-68c5-4ab7-8511-ca9cecdee3f0

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
**N/A**

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl: neomoth, themias
- add: Ability to rotate decals placed with crayons

neomoth committed:
implement crayon rotation

slarticodefast commented:
Duplicate of https://github.com/space-wizards/space-station-14/pull/34492
We will have to check which PR is better, but the other one also adds a preview.

neomoth committed:
add preview ghost

neomoth committed:
Update crayon UI, fix a few bugs.

fixed overlay not disappearing during most interactions where it leaves your hands. fixed a floating point rounding error. fixed issue where having more than one crayon would cause issues with the overlay displaying correctly

neomoth commented:
> I wish for preset angles (0, 45, 90, etc) but this works too

Sort of implemented now. Less so preset angles, more so giving more step intervals. Should be relatively easy to get the angle you want in just a few clicks now, and if something more precise is needed you can always just type it in the box.
Speaking of, I did some QOL stuff to that input field, tried my hardest to make so it wouldn't be annoying when dealing with stuff like negative numbers.

neomoth committed:
code cleanup pt1

DrSmugleaf commented on Content.Shared/Crayon/SharedCrayonComponent.cs:
```diff
@@ -21,6 +21,21 @@ public abstract partial class SharedCrayonComponent : Component
+        /// <summary>
+        /// Rotation of the resulting decal
+        /// </summary>
+        [DataField("rotation")]
+        public float Rotation;
+
+        [DataField("previewEnabled")]
+        public bool PreviewEnabled;
+
+        [DataField("previewVisible")]
+        public bool PreviewVisible;
+
+        [DataField("opaqueGhost")]
+        public bool OpaqueGhost;
```
Names don't need to be specified as long as they match the field name
```suggestion
        [DataField]
        public float Rotation;

        [DataField]
        public bool PreviewEnabled;

        [DataField]
        public bool PreviewVisible;

        [DataField]
        public bool OpaqueGhost;
```

neomoth committed:
Update Content.Shared/Crayon/SharedCrayonComponent.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

neomoth commented:
> Still being supported? 👉 👈

IDK dawg, this been on SL for months. Kinda gave up on this ever getting merged upstream.
